ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/HardFork/Combinator/InjectTxs.hs

```diff
@@ -92,6 +97,74 @@ matchPolyTx is tx =
         , currentState = Pair tx' currentState
         }
 
+
+matchPolyTxs' ::
+     All Top xs
+  => InPairs (InjectPolyTx tx) xs
+  -> [NS tx xs]
+  -> Telescope g f xs
+  -> ( [(NS tx xs, Mismatch tx f xs)]
+     , Telescope g (Product ([] :.: (,) (NS tx xs) :.: tx) f) xs
+     )
```

Comment:
I think using a cusotm type here would make it more clear what is happening

Reply:
Done